scapy/layers/ntp.py

```diff
@@ -394,6 +394,9 @@ def getfield(self, pkt, s):
             extensions_len = len(extensions)
             while extensions_len >= 16:
                 ext_len = struct.unpack("!H", extensions[2:4])[0]
+                ext_len = min(ext_len, extensions_len)
+                if ext_len < 1:
+                    ext_len = extensions_len
```

Comment:
You are right. I have added one regression test that should trigger an infinite loop with the current version of the code.